### PR TITLE
Unified error handling - Basic Messages

### PR DESCRIPTION
**Pull request by jonmattgray**
First step of using a single, unified error handler that catches errors thrown in controllers and sets the response and logs correctly.
Also fixes `npm run dev` and removes a couple of build files from the repo

Example 404
image
Logs
image

Example 500
image
Logs
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,8 +2,8 @@ config/local.json
 
 node_modules/
 build
-src/swagger.json
-src/routes.ts
```
Comment (jonmattgray) [inline .gitignore:6]: incorrect path before meant that build files were pushed to the repo

File: docker-compose-agent-ipfs.yml [hunk 2]
```diff
@@ -28,9 +27,6 @@ services:
     depends_on:
       ipfs0:
         condition: service_healthy
-    environment:
-      # possible to set values using env variables
-      AFJ_REST_LOG_LEVEL: 1
```
Comment (jonmattgray) [inline docker-compose-agent-ipfs.yml:33]: as far as I can tell this env does nothing and log level is set in config files e.g. `samples/cliConfig.json`

File: package-lock.json [hunk 3]
```diff
@@ -9107,15 +9094,6 @@
         "node": ">=12"
       }
     },
-    "node_modules/dynamic-dedupe": {
```
Comment (Ellenn-A) [inline package-lock.json:9110]: out of interest why are so many node_modules being removed here? were they used for the previous Error handling fixtures and we do not need them anymore?

File: src/error.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      })
+    }
+
+    if (err instanceof HttpResponse) {
+      logger.warn(`Error thrown in handler: ${err.message}`)
+
+      return res.status(err.code).json(err.message)
+    }
+    if (err instanceof Error) {
+      logger.error(`Unexpected error thrown in handler: ${err.message}`)
```
Comment (Ellenn-A) [inline src/error.ts:47]: in the original code it was like this:  
```
 agent.config.logger.error('Internal Server Error.', err)
      return res.status(500).json({
        message: 'Internal Server Error. Check server logging.',
      })
    }
```

I was just wondering do we no longer include 'Internal server error' we just say 'unexpected error in handler '? Are these two basically the same/interchangeable?

File: package-lock.json (hunk 3 above)
Comment (jonmattgray) [reply package-lock.json:9110]: I think they were used by `ts-node-dev`, which I removed as it was only used in the old `npm run dev` command (depcheck flagged it to me)

File: src/error.ts (hunk 4 above)
Comment (jonmattgray) [reply src/error.ts:47]: They seem interchangeable to me. The new method actually returns more info to the user, maybe too much cause it includes the stack
image

File: src/server.ts [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
 import { serve, generateHTML } from 'swagger-ui-express'
-import { ValidateError } from 'tsoa'
 import { container } from 'tsyringe'
 
 import { basicMessageEvents } from './events/BasicMessageEvents'
 import { connectionEvents } from './events/ConnectionEvents'
 import { credentialEvents } from './events/CredentialEvents'
 import { proofEvents } from './events/ProofEvents'
 import { RegisterRoutes } from './routes/routes'
+import { errorHandler } from './error'
```
Comment (msterle) [inline src/server.ts:18]: 👍  for putting this in its own module already

File: src/error.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      logger.trace(`Stack: ${err.stack}`)
+
+      return res.status(500).json(err)
+    }
+
+    next()
```
Comment (msterle) [inline src/error.ts:53]: should this be `next(err)` to propagate the error?

**Review by msterle — COMMENTED** [review]
looks like a great start

File: src/error.ts (hunk 6 above)
Comment (jonmattgray) [reply src/error.ts]: added

**Review by msterle — APPROVED** [review]
👍

**Pull request merged**